### PR TITLE
ARROW-10907: [Rust] Fix Cast UTF8 to Date64

### PR DESCRIPTION
**Pull request by seddonm1**
This PR fixes the behavior of a UTF8 -> Date64 conversion process to use `%Y-%m-%dT%H:%M:%S` rather than `%Y-%m-%d` with `00:00:00` time component.

It aligns with https://github.com/apache/arrow/pull/8913.

### CODE REVIEW

**Review by nevi-me — APPROVED** [review]
LGTM

File: rust/arrow/src/compute/kernels/cast.rs [hunk 1]
```diff
@@ -401,19 +401,18 @@ pub fn cast(array: &ArrayRef, to_type: &DataType) -> Result<ArrayRef> {
...
                     if string_array.is_null(i) {
                         builder.append_null()?;
                     } else {
-                        match NaiveDate::parse_from_str(string_array.value(i), "%Y-%m-%d")
-                        {
-                            Ok(date) => builder.append_value(
-                                date.and_time(zero_time).timestamp_millis() as i64,
-                            )?,
+                        match NaiveDateTime::parse_from_str(
+                            string_array.value(i),
```
Comment (Dandandan) [inline rust/arrow/src/compute/kernels/cast.rs:412]: See latest commits in #8913. There is a default parser that also avoids parsing the format for each value, and also supports the decimal fraction

Comment (seddonm1) [reply rust/arrow/src/compute/kernels/cast.rs:412]: Thanks @Dandandan will update.

Comment (seddonm1) [reply rust/arrow/src/compute/kernels/cast.rs]: have updated with your suggested changes. thank you very much - this helps me learn.

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @Dandandan  and @seddonm1 👍

**Pull request closed without merging**